### ISSUE TITLE
`prioritize` fails using demo data, with older versions of `vctrs`

### ISSUE THREAD

**Issue by jdhoffa** [2DegreesInvesting/r2dii.match#216]
I have just updated to R4.0.2 seem to have accidentally installed an old(ish) version of `vctrs`. 
The error seems to stem from passing the argument `relax = TRUE` to the function `vctrs::vec_proxy_compare()`. 
(Seems this argument did not exist in earlier versions of the `vctrs` pacakge). 

Confirmed that `prioritize()` works fine with (at least) version `vctrs > 0.3.1`

---

``` r
library(r2dii.data)
library(r2dii.match)

matched <- match_name(loanbook_demo, ald_demo)

prioritize(matched)
#> Error: `...` is not empty.
#> 
#> We detected these problematic arguments:
#> * `relax`
#> 
#> These dots only exist to allow future extensions and should be empty.
#> Did you misspecify an argument?
```

<sup>Created on 2020-06-26 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by maurolepore** [comment]
I could push an update to CRAN today. It'll be nice for everything to just work once r2dii.analysis is on CRAN.

**Comment by jdhoffa** [comment]
sure!

**Comment by maurolepore** [comment]
Can you repeat your reprex, this time with `si = TRUE`? r2dii.match does not use vctrs directly so I'll remove it from DESCRIPTION (not sure how it made it there) but likely via rlang. I suspect we should be fine with `rlang (>= 0.1.2)` (as we do in r2dii.analysis) but I would like to see what package-versions you have.

**Comment by maurolepore** [comment]
It would also help to see the traceback, maybe with `rlang::last_error()` -- so I can narrow down the scope of the problem.

**Comment by maurolepore** [comment]
For the record: Jackson said the error came from `dplyr::arrange_at()`. This package does not use vctrs directly, but dplyr does. So we could depend on the oldest version of dplyr where `dplyr::arrange_at()` does not throw this error.

**Comment by maurolepore** [comment]
I failed to reproduce this error but it's now clear we need to enforce dplyr >= 0.8.5. 

``` r
library(r2dii.data)
library(r2dii.match)

packageVersion("r2dii.data")
#> [1] '0.1.1'

packageVersion("r2dii.match")
#> [1] '0.0.2'

packageVersion("dplyr")
#> [1] '0.8.4'

match_name(loanbook_demo, ald_demo)
#> Error in filter_impl(.data, quo): attempt to bind a variable to R_UnboundValue
```

<sup>Created on 2020-06-29 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

<details>

<summary>Session info</summary>

``` r
sessionInfo()
#> R version 4.0.0 (2020-04-24)
#> Platform: x86_64-pc-linux-gnu (64-bit)
#> Running under: Debian GNU/Linux bullseye/sid
#> 
#> Matrix products: default
#> BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0
#> LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0
#> 
#> locale:
#>  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
#>  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
#>  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
#>  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
#>  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
#> [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
#> 
#> attached base packages:
#> [1] stats     graphics  grDevices utils     datasets  methods   base     
#> 
#> other attached packages:
#> [1] r2dii.match_0.0.2 r2dii.data_0.1.1 
#> 
#> loaded via a namespace (and not attached):
#>  [1] Rcpp_1.0.4.6       knitr_1.29         magrittr_1.5       tidyselect_1.1.0  
#>  [5] R6_2.4.1           rlang_0.4.6        stringr_1.4.0      highr_0.8         
#>  [9] dplyr_0.8.4        tools_4.0.0        parallel_4.0.0     xfun_0.15         
#> [13] htmltools_0.5.0    ellipsis_0.3.1     yaml_2.2.1         digest_0.6.25     
#> [17] assertthat_0.2.1   tibble_3.0.1       lifecycle_0.2.0    crayon_1.3.4      
#> [21] tidyr_1.1.0        purrr_0.3.4        vctrs_0.3.1        stringdist_0.9.5.5
#> [25] glue_1.4.1         evaluate_0.14      rmarkdown_2.3      stringi_1.4.6     
#> [29] compiler_4.0.0     pillar_1.4.4       pkgconfig_2.0.3
```

</details>

--

FYI I did this with 

````

reprex::reprex(si = TRUE, {
  library(r2dii.data)
  library(r2dii.match)
  
  packageVersion("r2dii.data")
  
  packageVersion("r2dii.match")
  
  packageVersion("dplyr")
  
  match_name(loanbook_demo, ald_demo)
})
````

**Comment by maurolepore** [comment]
Closed by #219
